src/bartiq/compilation/_utilities.py

```diff
@@ -92,13 +93,69 @@ def get_children_in_walk_order(routine: T) -> list[T]:
 
 def parse_value(value_str: str) -> Number:
     """Attempts to parse a single value string, but throws a compilation error if this isn't possible."""
+    value_str = value_str.strip()
+
+    # Check if it is a math constant
+    if value_str in Math_constants:
+        return Math_constants[value_str]
+
+    # Check if it is a numeric value
     try:
         value = ast.literal_eval(value_str)
-        if not isinstance(value, NUMBER_TYPES):
-            raise ValueError
-        return value
+        if isinstance(value, NUMBER_TYPES):
+            return value
+    except ValueError:
+        pass
+
+    # Check if it is a mathematical function call
+    try:
+        return evaluate_math_function(value_str)
```

Comment:
It seems to me that this logic shouldn't belong here.
Looking at the docstring, the intention of `parse_value` was to parse expressions representing a single value, and function call is not a single value.

Reply:
Removed.